app/js/reducers/artifacts.js

```diff
@@ -23,8 +23,11 @@ const artifacts = (state = initialState, action) => {
         });
         return state.set(action.artifact.id, nextState);
     }
-    case 'DELETE_ARTIFACT':
-        return state.delete(action.id);
+    case 'DELETE_ARTIFACT': {
+        // TODO, make it actually do something
+        const newState = ImmutableMap(state);
+        return newState.delete(action.id).toJS();
```

Comment:
Working on the switch, requires a pretty hefty rewrite of test expectations and reducer functions. Should I just close this request and do a new one when it is finished?